AgentDiscoveries-Backend/src/main/java/org/softwire/training/api/routes/v1/AgentsRoutes.java

```diff
@@ -14,19 +15,25 @@
 public class AgentsRoutes {
 
     private final AgentsDao agentsDao;
+    private final PermissionsVerifier permissionsVerifier;
 
     @Inject
-    public AgentsRoutes(AgentsDao agentsDao) {
+    public AgentsRoutes(AgentsDao agentsDao, PermissionsVerifier permissionsVerifier) {
         this.agentsDao = agentsDao;
+        this.permissionsVerifier = permissionsVerifier;
     }
 
     public Agent createAgent(Request req, Response res) throws FailedRequestException {
         Agent agentModel = JsonRequestUtils.readBodyAsType(req, Agent.class);
-
         if (agentModel.getAgentId() != 0) {
             throw new FailedRequestException(ErrorCode.INVALID_INPUT, "agentId cannot be specified on create");
         }
 
+        int userId = req.attribute("user_id");
+        if (!permissionsVerifier.isAdmin(userId)) {
+            throw new FailedRequestException(ErrorCode.OPERATION_INVALID, "user doesn't have valid permissions");

```

Comment:
Are you sure `OPERATION_INVALID` is correct? We want this to return a `403 Forbidden` to indicate that they are authenticated, but do not have permission to perform the action.